build/common.sh

```diff
@@ -1479,6 +1479,14 @@ function kube::release::docker::release() {
     docker_push_cmd=("gcloud" "docker")
   fi
 
+  if [[ "${KUBE_DOCKER_REGISTRY}" == "gcr.io/google_containers" ]]; then
+    if ! kube::release::has_gcloud_account k8s.production.user@gmail.com; then
+      kube::log::error "Pushing images to gcr.io/google_containers requires credentials for account k8s.production.user@gmail.com"
+      return 1
```

Comment:
Ideally we want to verify (all of the) auths at the very top of the pipeline/process and then fail as needed before we even start.
If we absolutely need to allow a handled exception in the middle of push, can we make sure the push tooling (command-line) itself is be re-entrant so it can simply be run again when authorization is available?  I think this may already be the case.